### DISCUSSION TITLE
How to retrieve L1 level held-out predictions on Cross Validation Folds

### DISCUSSION THREAD

**Discussion by hongzhoulin89** [autogluon/autogluon#1187] [Q&A] [answered]
<img width="713" alt="screen_shot" src="https://user-images.githubusercontent.com/29802555/122298168-478eb480-ceb1-11eb-8066-eb5b8e3eb136.png">

According to the AWS presentation https://pages.awscloud.com/Hyperparameter-Tuning-is-Overrated-Apply-AutoGluon-Tabular-an-AutoML-Algorithm-from-AWS-Marketplace_2020_0911-MKT_OD.html

AutoGluon held-out a validation set on each fold, those data are considered as cross validation as they are not included in the training data (more precisely, 1st validation set (white) is not included in Fold 1 training, 2nd validation set (green) is not included in Fold 2 training, etc) What I am looking for is exactly those validation prediction: the prediction on 1st validation set by model 1,   the prediction on 2nd validation set by model 2,  etc. In other words the right hand side column where the validation prediction are concatenated.  Is there a way to retrieve them from the predictor module?

**Comment by Innixma** [comment, marked as answer]
Yes, you can call `predictor.get_oof_pred_proba()`.

Refer to https://auto.gluon.ai/stable/api/autogluon.predictor.html#autogluon.tabular.TabularPredictor.get_oof_pred_proba

**Reply by hongzhoulin89** [reply]
Thanks, that’s exactly what I am looking for

**Comment by Vikram12301** [comment]
I am getting this error 
```
AssertionError: Predictor must be in bagged mode to get out-of-fold predictions.
```
